pkg/channel/distributed/controller/kafkachannel/reconciler.go

```diff
@@ -221,22 +228,26 @@ func (r *Reconciler) reconcile(ctx context.Context, channel *kafkav1beta1.KafkaC
 
 // configMapObserver is the callback function that handles changes to our ConfigMap
 func (r *Reconciler) configMapObserver(configMap *corev1.ConfigMap) {
-	if configMap == nil {
-		r.logger.Warn("Nil ConfigMap passed to configMapObserver; ignoring")
+
+	if r == nil {
+		// This typically happens during startup and can be ignored
 		return
 	}
-	if r == nil {
-		// This typically happens during startup
-		r.logger.Debug("Reconciler is nil during call to configMapObserver; ignoring changes")
+
+	if configMap == nil {
+		r.logger.Warn("Nil ConfigMap passed to configMapObserver; ignoring")
 		return
 	}
 
+	// Add the config map to the logger for assistance with debug, error and fatal diagnostics
+	logger := r.logger.With(zap.Any("configMap", configMap))
+
 	// Enable Sarama Logging If Specified In ConfigMap
 	if ekConfig, err := kafkasarama.LoadEventingKafkaSettings(configMap); err == nil && ekConfig != nil {
 		kafkasarama.EnableSaramaLogging(ekConfig.Kafka.EnableSaramaLogging)
-		r.logger.Debug("Updated Sarama logging", zap.Bool("Kafka.EnableSaramaLogging", ekConfig.Kafka.EnableSaramaLogging))
+		logger.Debug("Updated Sarama logging", zap.Bool("Kafka.EnableSaramaLogging", ekConfig.Kafka.EnableSaramaLogging))
 	} else {
-		r.logger.Error("Could Not Extract Eventing-Kafka Setting From Updated ConfigMap", zap.Error(err))
+		logger.Error("Could Not Extract Eventing-Kafka Setting From Updated ConfigMap", zap.Error(err))
 	}
 
```

Comment:
Is this if/else correct?  Seems like it will ALWAY log an error?

Reply:
I haven't changed the logic here; it should only log an error if either "err != nil" or "ekConfig == nil" according to the if condition.  Enabling and disabling the sarama logging via the configmap definitely works (or at least it worked when I tested that feature in a previous PR).

Reply:
👍  I read it wrong before - looks ok - thx/sorry ; )